### DISCUSSION TITLE
Connecting to service connected to gluetun?

### DISCUSSION THREAD

**Discussion by mdf135** [qdm12/gluetun#399] [Q&A]
I'm not sure what I am doing wrong, it seems like I should be able to put in localhost:8080 and get to the qBittorrent web UI. Am I missing something obvious here? I still fairly new with docker-compose and newer still will anything VPN related.

I'm running this on a raspberry pi 4.

Here is my docker-compose.yml (changed to .txt so I could attach it); 
[docker-compose.txt](https://github.com/qdm12/gluetun/files/6087923/docker-compose.txt)

My logs from gluetun; 
[gluetun.log](https://github.com/qdm12/gluetun/files/6087927/gluetun.log)

My logs from qBittorrent; 
[qbittorrent.log](https://github.com/qdm12/gluetun/files/6087929/qbittorrent.log)

**Comment by raph521** [comment]
Are you on a PC and trying to access your Pi over your local network?  If so, then in your PC's browser, you'd need `http://<ip-of-pi>:8080`

**Reply by mdf135** [reply]
Thank you for the response. I am accessing the pi over the local network and I am using the local ip of the pi, I haven't had any issues connecting to the web UIs of other applications I'm running on the pi. I ended up breaking up the docker-compose into two, so the gluetun was one docker-compose and qBittorrent was another. I connected to gluetun using 'network_mode: container:gluetun" and it is working fine now.

While I was separating them I did notice that I had lost the "restart: always" in the gluetun service section of the compose. I don't imagine that would cause the issue but it was the only thing I noticed and added.

**Reply by qdm12** [reply]
So using `container:gluetun` solves it but it doesn't work with `service:gluetun`?

To me your docker-compose.yml looks fine.

**Reply by mdf135** [reply]
That is correct, or at least I couldn't get it to work with `service:gluetun`. Thank you for looking over my docker-compose.yml.

**Reply by qdm12** [reply]
Maybe does it work with `network_mode: "service:gluetun"` using double quotes? That's what I use in my docker-compose.yml 🤔

**Reply by mdf135** [reply]
I believe that is how I had it originally, since that was how you had it in your examples, but I changed it when I couldn't get it to work, as the without double quotes had worked for me in a different vpn container. However, I'm not sure so I'll give it a shot when I get some time this weekend. Thank you for the suggestions.

**Reply by raph521** [reply]
The only thing I can think of is perhaps the gluetun service was restarted?  I think that if you restart gluetun, you need to restart (or maybe even re-create) the dependent services, in this case qbittorrent.

Defining the services in the same docker-compose should be fine, that's what I do, and I don't use double quotes in my docker-compose.yml
